lv2/sfizz.c

```diff
@@ -975,8 +968,8 @@ restore(LV2_Handle instance,
 
         lv2_log_note(&self->logger, "[sfizz] Restoring the file %s\n", path);
         // Load an empty file to remove the default sine
-        sfizz_lv2_get_empty_sfz_path(instance, self->sfz_file_path, MAX_PATH_SIZE);
-        sfizz_lv2_load_file(instance, self->sfz_file_path);
+        strcpy(self->sfz_file_path, "");
```

Comment:
There seems to be an inconsistency about this decision to reset here.
The sfz gets reset, but not the scala down below.

Moreover it's precisely what this implementation would attempt to avoid before; always keep a non-empty string in there, to work around the problem with Ardour.

The string remains empty if the following load fails.
Perhaps we could store the paths, regardless if the file loading succeeds or not? (+ it's ensuring plugin and host parameters side to be in sync)

Reply:
We can reset the scala too.

Well, this keeps the "behavior" of the previous version. Ardour will through state.ttl at us but we won't make sounds, so previously empty instances remain empty.

You're suggesting that we use the `state.ttl` file as the plugin path value in this case? It might be an indication for the user that something is wrong, so it's not a bad solution. In this case we need to change the `sfz_file_path` variable regardless of whether loading works.

Reply:
We can reset absolutely everything. `restore()` could be as
- set all values to defaults
- extract the parameters present in the object into values
- sync all values into the synth